src/child/proptypes/favourites.js

```diff
@@ -0,0 +1,8 @@
+import { PropTypes } from 'react';
+
+const favouritesShape = PropTypes.shape({
+    codes: PropTypes.array.isRequired,
+    names: PropTypes.object.isRequired
+}).isRequired;
```

Comment:
Minor detail and just a thought - it might be a bit more flexible and I'm not sure how clear it is when using these shapes that they are required - but the `.isRequired` could be omitted here.

E.g.

**Before**

```js
const favouritesShape = PropTypes.shape({
    codes: PropTypes.array.isRequired,
    names: PropTypes.object.isRequired
}).isRequired;

Whatever.propTypes = {
    favourites: favouritesShape
};
```
**After**

```js
const favouritesShape = PropTypes.shape({
    codes: PropTypes.array.isRequired,
    names: PropTypes.object.isRequired
});

Whatever.propTypes = {
    favourites: favouritesShape.isRequired
};
```